src/main.cpp

```diff
@@ -735,15 +735,22 @@ void handleMachineState() {
         case kPidNormal:
             brewDetection();
 
-            if ((timeBrewed > 0 && FEATURE_BREWCONTROL == 0) || (FEATURE_BREWCONTROL == 1 && currBrewState > kBrewIdle && currBrewState <= kBrewFinished)) {
+            #if (FEATURE_BREWCONTROL == 1)
+                brew(); 
+            #endif
+
+            checkHotWaterSwitch();
+            checkSteamSwitch();
+
+            if (((timeBrewed > 0 && FEATURE_BREWCONTROL == 0) || (FEATURE_BREWCONTROL == 1 && currBrewState > kBrewIdle && currBrewState <= kBrewFinished)) && hotWaterOn == 0 && steamON == 0) {
```

Comment:
this is way to verbose and not needed:
1. switch checks should only happen if feature is enabled -> Feature flags for brew, backflush, hotwater, steam
2. it is not nessesary to check if there is no brew running before go to e.g. hotWater. -> if brew is running the state mashine is in kBrew
3. brew() should also not be needed, this is handled via brewDetection()